tests/services/test_course_service.py

```diff
@@ -0,0 +1,94 @@
+import unittest
+import httpx
+from unittest.mock import MagicMock
+from sqlalchemy.orm import Session
+from sqlalchemy.orm.exc import MultipleResultsFound, NoResultFound
+from app.services import CourseService
+from app.models import CourseModel, InstructorModel
+from app.services.user import InstructorService
+from app.services import GiteaService
+from ..data.database.course import data
+from app.schemas import CourseWithInstructorsSchema
+from app.core.exceptions import (
+    MultipleCoursesExistException,
+    NoCourseExistsException,
+    CourseAlreadyExistsException
+)
+
+class TestCourseService(unittest.IsolatedAsyncioTestCase):
+    async def asyncSetUp(self):
+        self.mock_session = MagicMock(spec=Session)
+
+        self.course_service = CourseService(session=self.mock_session)
+        self.mock_course = CourseModel(name="COMP 555", master_remote_url="")
+
+    async def test_get_course_single_result(self):
+        self.mock_session.query().one.return_value = self.mock_course
+        result = await self.course_service.get_course()
+
+        self.assertIsInstance(result, CourseModel)
+        self.assertEqual(result, self.mock_course)
+    
+    async def test_get_course_multiple_results(self):
+        self.mock_session.query().one.side_effect = MultipleResultsFound()
+
+        with self.assertRaises(MultipleCoursesExistException):
+            self.assertRaises(await self.course_service.get_course())
+
+    async def test_get_course_no_result(self):
+        self.mock_session.query().one.side_effect = NoResultFound()
+
+        with self.assertRaises(NoCourseExistsException):
+            self.assertRaises(await self.course_service.get_course())
+    
+    async def test_get_course_with_instructors_schema_success(self):
+        mock_course = CourseModel(id=1, name="Math", master_remote_url="http://example.com")
+        instructorList = [InstructorModel(id=1, onyen="instructor_onyen", first_name="Ins", last_name="Tructor", email="email@unc.com")]
+
+        self.mock_session.query().one.return_value = mock_course
+        self.mock_session.query().all.return_value = instructorList
+
+        expected_course = CourseModel(id=1, name="Math", master_remote_url="http://example.com")
+        expected_course.instructors=instructorList
+
+        result = await self.course_service.get_course_with_instructors_schema()
+        expected_result = CourseWithInstructorsSchema.from_orm(expected_course)
+
+        self.assertEqual(result, expected_result)
+
+    async def test_get_instructor_gitea_organization_name(self):
+        self.mock_session.query().one.return_value = self.mock_course
+
+        result = await self.course_service.get_instructor_gitea_organization_name()
+        assert result == "COMP 555-instructors"
```

Comment:
I tend not to like "magic numbers" in code, which "COMP 555" essentially is here since it's not clear where that comes from. If it's easy to just create a var at the top like `course_name = "COMP 555"` and use `course_name` wherever you need "COMP 555" that would be ideal, but it's not too big of a deal. It's not thaaat likely that someone will want to change this value in the future, so 🤷

Reply:
Yea on second thought I don't think this is necessary at all